numba/rewrites/static_getitem.py

```diff
@@ -21,19 +21,29 @@ def match(self, interp, block, typemap, calltypes):
                     const = interp.infer_constant(expr.index)
                 except errors.ConstantInferenceError:
                     continue
-                getitems.append((expr, const))
+                getitems[expr] = const
 
         return len(getitems) > 0
 
     def apply(self):
         """
         Rewrite all matching getitems as static_getitems.
         """
-        for expr, const in self.getitems:
```

Comment:
This pass introduce in opcode to the source IR.  It is rewritten to create new IR node instead.  We should consider refactoring all rewrite passes to avoid mutating the source IR.